packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/recent-workspace.tsx

```diff
@@ -0,0 +1,119 @@
+import classnames from 'classnames';
+import React from 'react';
+import { Icon } from '@deriv/components';
+import { timeSince } from '@deriv/bot-skeleton';
+import { save_types } from '@deriv/bot-skeleton/src/constants/save-type';
+import { connect } from 'Stores/connect';
+import RootStore from 'Stores/index';
+import './index.scss';
+
+type TRecentWorkspace = {
+    getRecentFileIcon: (string: string) => void;
+    getSaveType: () => void;
+    previewRecentStrategy: () => void;
+    selected_strategy_id: string;
+    workspace: { [key: string]: string };
+    setFileLoaded: (param: boolean) => void;
+    index: number;
+    has_file_loaded: boolean;
+    toggleSaveModal: () => void;
+    toggleDeleteDialog: () => void;
+    loadFileFromRecent: () => void;
+    setActiveTab: (active_tab: number) => void;
+};
+
+const RecentWorkspace = ({
+    getRecentFileIcon,
+    getSaveType,
+    previewRecentStrategy,
+    selected_strategy_id,
+    workspace,
+    index,
+    has_file_loaded,
+    setFileLoaded,
+    toggleDeleteDialog,
+    toggleSaveModal,
+    loadFileFromRecent,
+    setActiveTab,
+}: TRecentWorkspace) => {
+    const trigger_div_ref = React.useRef<HTMLInputElement | null>(null);
+    React.useEffect(() => {
+        if (index === 0 && has_file_loaded === false) {
+            trigger_div_ref?.current?.click();
+        }
+    }, [has_file_loaded]);
+    return (
+        <>
+            <div
+                className={classnames('load-strategy__recent-item', {
+                    'load-strategy__recent-item--selected': selected_strategy_id === workspace.id,
+                })}
+                ref={trigger_div_ref}
+                key={workspace.id}
+                onClick={
+                    selected_strategy_id === workspace.id
+                        ? undefined
+                        : () => {
+                              previewRecentStrategy(workspace.id);
+                              setFileLoaded(true);
+                          }
+                }
+            >
+                <div className='load-strategy__recent-item-text'>
+                    <div className='load-strategy__recent-item-title'>{workspace.name}</div>
+                </div>
+                <div className='load-strategy__recent-item-time'>{timeSince(workspace.timestamp)}</div>
+                <div className='load-strategy__recent-item-location'>
+                    <Icon
+                        icon={getRecentFileIcon(workspace.save_type)}
+                        className={classnames({
+                            'load-strategy__recent-icon--active':
+                                true || workspace.save_type === save_types.GOOGLE_DRIVE,
```

Comment:
in this condition ` true || workspace.save_type === save_types.GOOGLE_DRIVE` the first one is always true and never reaches the second condition .. right ?

Reply:
yes this was legacy code removed the first one and kept the second condition